ion-schema/src/isl/isl_constraint.rs

```diff
@@ -377,6 +383,69 @@ impl IslConstraint {
             "timestamp_precision" => Ok(IslConstraint::TimestampPrecision(
                 Range::from_ion_element(value, RangeType::TimestampPrecision)?,
             )),
+            "timestamp_offset" => {
+                use IonType::*;
+                if value.is_null() {
+                    return invalid_schema_error(
+                        "expected a list of valid offsets for an `timestamp_offset` constraint, found null",
+                    );
+                }
+
+                if value.annotations().next().is_some() {
+                    return invalid_schema_error("`timestamp_offset` list may not be annotated");
+                }
+
+                let valid_offsets: Vec<TimestampOffset> = match value.ion_type() {
+                    List => {
+                        let list_values = value.as_sequence().unwrap();
+                        if list_values.is_empty() {
+                            return invalid_schema_error(
+                                "`timestamp_offset` constraint must contain at least one offset",
+                            );
+                        }
+                        let list_vec: IonSchemaResult<Vec<TimestampOffset>> = list_values
+                            .iter()
+                            .map(|e| {
+                                if e.is_null() {
+                                    return invalid_schema_error(
+                                    "`timestamp_offset` values must be non-null strings, found null"
+                                );
+                                }
+
+                                if e.ion_type() != IonType::String {
+                                    return invalid_schema_error(format!(
+                                    "`timestamp_offset` values must be non-null strings, found {}",
+                                    e
+                                ));
+                                }
+
+                                if e.annotations().next().is_some() {
+                                    return invalid_schema_error(format!(
+                                        "`timestamp_offset` values may not be annotated, found {}",
+                                        e
+                                    ));
+                                }
+
+                                // unwrap here will not panic as we have already verified the ion type to be a string
+                                let string_value = e.as_str().unwrap();
+
+                                // convert the string to TimestampOffset which stores offset in minutes
+                                string_value.try_into()
+                            })
+                            .collect();
+                        list_vec?
+                    }
+                    _ => {
+                        return invalid_schema_error(format!(
+                            "ion type: {:?} is not supported with occurs constraint",
```

Comment:
`IonType` implements `Display`. Constraint should be `timestamp_offset`.
```suggestion
                            "ion type: {} is not supported with timestamp_offset constraint",
```

That being said, how about a message that describes what the correct format is so that the user knows how to correct their mistake, like `"timestamp_offset requires a list of offset strings, but found: {value}"`?